tools/mason/MasonNew.chpl

```diff
@@ -67,64 +92,66 @@ proc masonNew(args) throws {
   }
 }
 
-proc validatePackageName(name) throws {
-  if name == '' {
+proc validatePackageName(dirName) throws {
+  if dirName == '' {
     throw new owned MasonError("No package name specified");
   }
-  else if !isIdentifier(name) {
-    throw new owned MasonError("Bad package name '" + name +
-                        "' - only Chapel identifiers are legal package names");
+  else if !isIdentifier(dirName) {
+    throw new owned MasonError("Bad package name '" + dirName +
+                        "' - only Chapel identifiers are legal package names.\n" +  
+                        "Please use mason new <illegal-name> --name <legal-name>");
   }
-  else if name.count("$") > 0 {
-    throw new owned MasonError("Bad package name '" + name +
+  else if dirName.count("$") > 0 {
+    throw new owned MasonError("Bad package name '" + dirName +
                         "' - $ is not allowed in package names");
   }
   else {
     return true;
   }
 }
 
-proc InitProject(name, vcs, show) throws {
+proc InitProject(dirName, packageName, vcs, show) throws {
   if vcs {
-    gitInit(name, show);
-    addGitIgnore(name);
+    gitInit(dirName, show);
+    addGitIgnore(dirName);
   }
   else {
-    mkdir(name);
+    mkdir(dirName);
   }
   // Confirm git init before creating files
-  if isDir(name) {
-    makeBasicToml(name, path=name);
-    makeSrcDir(name);
-    makeModule(name, fileName=name);
-    makeTestDir(name);
-    makeExampleDir(name);  
-    writeln("Created new library project: " + name);
+  if isDir(dirName) {
+    if dirName!=packageName then makeBasicToml(packageName, path=dirName);
```

Comment:
We shouldn't need to check if they are equal. If they are then, the calls are the same:

```suggestion
   makeBasicToml(packageName, path=dirName);
```